crates/subspace-farmer/src/archiving.rs

```diff
@@ -304,13 +313,15 @@ fn spawn_archiving(
                         }
                     };
 
+                    // The only case when broadcast channel might fail is when no-one is listening
```

Comment:
I think this isn't the case. This piece of logic must be racy and if one segment is archived before subscribers are created this will break too, which seems very problematic and will be hard to debug.